Applications/spire/include/spire/book_view/book_view_window.hpp

```diff
@@ -59,36 +60,29 @@ namespace Spire {
     protected:
       void closeEvent(QCloseEvent* event) override;
       bool eventFilter(QObject* watched, QEvent* event) override;
-      void keyPressEvent(QKeyEvent* event) override;
 
     private:
-      mutable ChangeSecuritySignal m_change_security_signal;
       mutable ClosedSignal m_closed_signal;
       BookViewProperties m_properties;
       SecurityInputModel* m_input_model;
       std::shared_ptr<BookViewModel> m_model;
-      SecurityStack m_securities;
-      Nexus::Security m_current_security;
+      SecurityWidget* m_security_widget;
       QWidget* m_body;
+      QWidget* m_container_widget;
       QVBoxLayout* m_layout;
       TechnicalsPanel* m_technicals_panel;
       std::unique_ptr<BboQuotePanel> m_bbo_quote_panel;
-      std::unique_ptr<QWidget> m_overlay_widget;
-      std::unique_ptr<QLabel> m_empty_window_label;
       std::unique_ptr<TransitionWidget> m_transition_widget;
+      QTimer m_transition_timer;
```

Comment:
Why is this timer part of the ```BookViewWindow``` instead of the ```TransitionWidget```? This will require every window that wants a loading screen to have to manage its own timer which defeats the purpose of factoring out all of the common functionality into the ```TransitionWidget``` class.

Find a way to implement this without every window managing its own ```QTimer```.

Reply:
I can add the timer to TransitionWidget and have a callback for when the timer has been triggered, but I can't completely remove exposure of the windows to the timer because there's logic in the window that determines whether the TransitionWidget is shown or not.

Reply:
Can you describe that in more detail? As far as I know a window should show a TransitionWidget as soon as a new model has been assigned to it and remove the TransitionWidget as soon as the model is finished loading. What other logic is there?